packages/snap-controller/src/Search/SearchController.ts

```diff
@@ -110,25 +110,49 @@ export class SearchController extends AbstractController {
 			const stringyParams = JSON.stringify(storableRequestParams);
 
 			if (this.config.settings?.infinite?.restorePosition) {
-				// restore the scroll position saved previously
-				const scrollMap = this.storage.get('scrollMap') || {};
-
-				// interval we ony need to keep checking until the page height > than our stored value
-				const scrollToPosition = scrollMap[stringyParams];
-
-				if (scrollToPosition) {
-					let checkCount = 0;
-					const heightCheck = window.setInterval(() => {
-						if (document.documentElement.scrollHeight >= scrollToPosition) {
-							window.scrollTo(0, scrollToPosition);
-							this.log.debug('scrolling to: ', scrollMap[stringyParams]);
-							window.clearInterval(heightCheck);
-						}
-						if (checkCount > 2000 / HEIGHT_CHECK_INTERVAL) {
-							window.clearInterval(heightCheck);
-						}
-						checkCount++;
-					}, HEIGHT_CHECK_INTERVAL);
+				if (this.config.settings?.infinite.selector) {
```

Comment:
Maybe some of this code could be DRYed out a bit?

Lots of shared code here that I think could be simplified a bit.